doc/design.rst

```diff
@@ -0,0 +1,134 @@
+===============
+Software Design
+===============
+
+.. contents:: `Contents`
+    :depth: 3
+    :local:
+
+---------------------------
+Design for mlx.traceability
+---------------------------
+
+.. item:: DESIGN-TRACEABILITY Top level design for mlx.traceability
+    :fulfills: RQT-TRACEABILITY
+
+    Since the sphinx documentation system is python based, and it allows for plugin development in seperate
+    python packages, python was chosen. No limit should exist on the version of python or sphinx.
+
+.. item:: DESIGN-ITEMIZE Allow splitting the documentation in parts
+    :depends_on: DESIGN-TRACEABILITY
+    :fulfills: RQT-ITEMIZE
+
+    A directive name `item` is added to sphinx through the plugin that allows for splitting the documentation
+    into parts. The documentation parts are stored as objects of class `TraceableItem`. All `TraceableItem`
+    objects are stored in a container class `TraceableCollection`.
+
+.. item:: DESIGN-DOCUMENTATION_ID Identification of documentation part
+    :depends_on: DESIGN-ITEMIZE
+    :fulfills: RQT-ITEMIZE
+
+    A first argument to the `item` directive is used as a unique identifier for the documentation part. The
+    identifier can be any string - not containing spaces.
```

Comment:
Add here that it needs to be lowercase as well, otherwise it is automatically converted to lower case.

Reply:
no that lowercase is not in this branch, it is in one of the next ones. I will add it to docs on that branch.